src/main/scala/ai/privado/utility/Utilities.scala

```diff
@@ -617,6 +617,7 @@ object Utilities {
     randomSlug: String
   ): Unit = {
     builder.addEdge(impactedObject, originalSourceNode, EdgeTypes.ORIGINAL_SOURCE)
+    builder.addEdge(originalSourceNode, impactedObject, EdgeTypes.DERIVED_SOURCE)
     storeForTag(builder, originalSourceNode, ruleCache)(
       InternalTag.ORIGINAL_SOURCE_FOR_DERIVED_NODE.toString,
       Constants.originalSource + Constants.underScore + randomSlug
```

Comment:
If we want to add a tag for tracking purpose, let's add only the `name` as above and remove the value of `Constants.originalSource + Constants.underScore + randomSlug` as that doesn't serve as purpose

Reply:
We can remove the tag now that we have the edge to and from the node.